[DDJ-SB2] Fixes auto tempo going to infinity bug.

mab122 opened the pull request:
https://bugs.launchpad.net/mixxx/+bug/1856403/

When auto tempo adjustment is turned on it cannot be turned off.
That is because the code only disables it when `rate` is equal to 0.
But it only executes above check when variable `speed`(== `rate`) is DIFFERENT than 0.

Holzhaus commented:
Hey @mab122, thanks for your contribution. While you're at it, can you fix the Code Style issues that were discovered by our CI? https://travis-ci.org/github/mixxxdj/mixxx/jobs/663696708#L1221
You can fix most of them automatically by using [pre-commit](https://github.com/mixxxdj/mixxx/blob/master/.pre-commit-config.yaml#L20-L40).

Just install `pre-commit` and then run:
```
$ pre-commit run --files res/controllers/Pioneer-DDJ-SB2-scripts.js
```

Also, I guess that this bug is also present in the current Mixxx 2.2.3 release, right? That means that this bug should be fixed in the next bugfix release (2.2.4), not the next major release (2.3.0). Unfortunately, you filed this PR against the `master` branch (which is used for 2.3.0 development), not the `2.2` branch. Can you please rebase on `2.2` after fixing the linter issues? See https://mixxx.org/wiki/doku.php/using_git#targeting_another_base_branch for details how to do that.

mab122 committed:
Fixes auto tempo going to infinity bug

When auto tempo adjustment is turned on it cannot be turned off.
lp:1856403
Also this change marks unused variables (no more warnings by eslinter)

Holzhaus commented on res/controllers/Pioneer-DDJ-SB2-scripts.js:
```diff
@@ -492,34 +492,34 @@ PioneerDDJSB2.autodjToggle = function(channel, control, value, status, group) {
 
     if (PioneerDDJSB2.invertVinylSlipButton) {
-        controlsToFunctions.slip_enabled = 'PioneerDDJSB2.slipLed';
+        controlsToFunctions.slip_enabled = "PioneerDDJSB2.slipLed";
```
```suggestion
        controlsToFunctions.slipEnabled = "PioneerDDJSB2.slipLed";
```

mab122 committed:
Fixes `snakeCase` for few variables

Holzhaus reviewed (approved):
Looks good to me, thank you very much.
